src/core/continuous_space.jl

```diff
@@ -132,27 +132,27 @@ function biggest_id(model::ABM{A, <: ContinuousSpace}) where {A}
 end
 
 function randompos(space::ContinuousSpace)
-  pos = Tuple(rand(model.space.D))
-  !isnothing(model.space.extend) && (pos .*= model.space.extend)
+  pos = Tuple(rand(space.D))
+  !isnothing(space.extend) && (pos = pos .* space.extend)
   return pos
 end
 
-function add_agent!(agent::A, model::ABM{A, <: ContinuousSpace}) where {A}
-  agent.pos = randompos(model.space)
+function add_agent!(agent::A, model::ABM{A, <: ContinuousSpace}) where {A<:AbstractAgent}
+  # agent.pos = randompos(model.space) # Why not use the agent's position?

```

Comment:
Please read the public API of `agent_pos!`. The function that uses the agent's position is `add_agent_pos!`

Reply:
(this change has to be reverted for consistent API)